### PR TITLE
[Hub Generated] Review request for Microsoft.StorageCache to add version stable/2021-03-01

### PR DESCRIPTION
**Pull request by brpanask**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/brpanask/azure-rest-api-specs/dev-storagecache-Microsoft.StorageCache-2021-03-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by brpanask** [comment]
There's a customer expecting these feature updates in March, so it would be good if we can have the review approved and ready to merge soon.  It doesn't have an ARM reviewer yet and its been open for 14 days.

**Comment by markcowl** [comment]
@brpanask why is this marked as 'DoNotMerge' if it is an urgent customer ask?  S360 items are for fixing reported S360 completeness and correctness issues, not for providing new customer features.

**Comment by brpanask** [comment]
> @brpanask why is this marked as 'DoNotMerge' if it is an urgent customer ask? S360 items are for fixing reported S360 completeness and correctness issues, not for providing new customer features.

The RP still needs to roll this out and the ARM Manifest for the rollout needs to completed before this is merged and available to customer.  We also need it approved in case there are review comments.  So I have to start the review and collect comments that may need to go into the RP before its rolled out to production, but block it from merging once its approved until the rollout is complete.  Is there some other way to go about that?

The new version of the api contains new features and fixes.
There was a separate PR in the older versions for the s360 errors and this was created with the baseline copy from an older version before those s360 issues were merged.

Let me know if you need more detail or if thats not clear.
Thanks.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json [hunk 1]
```diff
@@ -554,18 +514,66 @@
         "operationId": "Caches_Update"
       }
     },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.StorageCache/caches/{cacheName}/storageTargets/{storageTargetName}/dnsRefresh": {
```
Comment (RamyasreeChakka) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:517]: nit: POST action names are generally verbs. IMO, name 'refreshDns' is apt here. I will leave the call to you.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json [hunk 2]
```diff
@@ -1461,10 +1387,26 @@
             "type": "string"
           },
           "type": "array"
+        },
+        "dnsServers": {
+          "description": "DNS servers for the cache to use. The default is determined by the network configuration.",
```
Comment (RamyasreeChakka) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1392]: What does 'default is determined by the network configuration' mean? Say, if user doesn't provide dnsServers values during PUT, would the default values be returned by RP in GET? If so please be aware that PUT & GET mismatch and you will run into issues in What-If testing. The PUT should be as close as possible to GET payload.
Refer https://armwiki.azurewebsites.net/api_contracts/WhatIfNoiseFix.html#put-as-patch-properties and  https://armwiki.azurewebsites.net/api_contracts/WhatIfNoise.html

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json [hunk 3]
```diff
@@ -1795,6 +1743,28 @@
         "statusDescription": {
           "description": "Describes explanation of state.",
           "type": "string"
+        },
+        "conditions": {
```
Comment (RamyasreeChakka) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1747]: Is this a readOnly property? If yes, please mark it as readonly.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json [hunk 4]
```diff
@@ -1795,6 +1743,28 @@
...
+            "$ref": "#/definitions/Condition"
+          }
+        }
+      }
+    },
+    "Condition": {
+      "type": "object",
+      "description": "Outstanding conditions that will need to be resolved.",
+      "properties": {
+        "timestamp": {
```
Comment (RamyasreeChakka) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1760]: Sounds like timestamp and message are readonly properties? If yes, please mark them as readonly.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json [hunk 5]
```diff
@@ -1994,25 +1951,30 @@
...
+    "BlobNfsTarget": {
+      "type": "object",
+      "description": "Properties pertaining to the BlobNfsTarget.",
+      "properties": {
+        "target": {
+          "description": "Resource ID of the storage container.",
+          "$ref": "#/definitions/URLString"
+        },
+        "usageModel": {
+          "description": "Identifies the usage model to be used for this Storage Target. Get choices from .../usageModels",
```
Comment (RamyasreeChakka) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1973]: Description text 'Get choices from .../usageModels' is not intuitive. Recommend to re-word it as these text will be published in docs.

Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json]: Updated in latest commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json (hunk 4 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1761]: Updated in latest commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json (hunk 3 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:1747]: Updated in latest commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json (hunk 5 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json]: Also updated an existing description with a similar message.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json (hunk 2 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json]: I tried to reword this but I'm not sure how much else to add or how to phrase it.
With DNS, DHCP or AD or other services on the network can provide the DNS configuration.  In Azure I think virtual networks have a variety of options for setting up dns servers on virtual networks which could also provide the configuration.   Whatever the network provides will be sent back in a GET and can be supplied in a PUT again or left out of a PUT.  Either way would work.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json (hunk 1 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2021-03-01/storagecache.json:517]: I discussed with others on the team and they requested that I keep it as is, so I did not update it here.

**Comment by chiragg4u** [comment]
Thank you for copying the existing version into new folder for first commit. It helps in reviewing the change. Marked as ARMApproved,

**Review by chiragg4u — APPROVED** [review]
:shipit:

**Comment by brpanask** [comment]
@markcowl
The RP side of things is ready and this can be merged now.

**Review by markcowl — APPROVED** [review]

**Pull request merged**